core/src/main/java/ai/timefold/solver/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -913,6 +901,74 @@ public boolean hasNoNullVariables(Object entity) {
         };
     }
 
+    private Predicate<Object> forEachIncludingPredicateBasicVarSupplier(EnumSet<ForEachInclude> includeSet) {
+        // TODO: Use a supply for this?
+        final var inconsistentDescriptor = this.getShadowVariablesInconsistentDescriptor();
+        final Predicate<Object> isUnassignedPredicate = this::hasNoNullVariables;
+        final Predicate<Object> isInconsistentPredicate = (inconsistentDescriptor == null) ? ignored -> true
+                : entity -> Boolean.FALSE.equals(inconsistentDescriptor.getValue(entity));
```

Comment:
Whether or not the entity is consistent is a property of the entity, and it exists regardless of any external annotation.

With these changes to `forEach`, using this annotation became pretty much unnecessary. (Unless the user actually needs to export inconsistent solutions, and therefore needs to have the information in their model.) Therefore this operation needs to work correctly even without the annotation being declared.

Arguably the answer to the question on line 905 is "yes". Since we still have plenty of time before the release, I suggest we do this in this PR.

The docs also need to be updated to reflect the new role of the annotation. (This probably also touches the CS docs, which would now need backlinks to the chapter about DSV.)